### PR TITLE
Docker and prisma

### PR DESCRIPTION
**Pull request by Frankdias92**
### Overview of Changes

This pull request introduces the following changes and new files to the project:

1. **`docker-compose.yml`**  
   - Added a `docker-compose.yml` file to define services and manage containers.
   - Configured a PostgreSQL service (`sippy-api-pg`) using the Bitnami PostgreSQL image, with default credentials (`docker` as both username and password) and the database name `apisippy`.
   - Configured persistent storage with a volume named `pgdata` to ensure data is saved across container restarts.

2. **`package.json` and `package-lock.json`**  
   - Added Prisma-related scripts:
     - `"prisma:generate"` to generate Prisma Client.
     - `"prisma:migrate"` to create migrations during development.
     - `"prisma:deploy"` to apply migrations in production.
   - Added dependencies:
     - `@prisma/client`, `dotenv`, `pg` for Prisma and PostgreSQL integration.
     - Updated the project to use `"type": "module"` to enable ES module syntax.

3. **Prisma Migration Files**  
   - Added a migration under `prisma/migrations/` which includes the following tables:
     - `users`: Stores user information with fields like `id`, `isAdmin`, `email`, etc.
     - `resident_requests`: Stores requests made by users with status (e.g., PENDING, COMPLETED).
     - `addresses`: Stores address information linked to users.
     - `time_slots`: Stores available time slots for scheduling.
   - Created various foreign key constraints to establish relationships between tables.

4. **Prisma Schema (`schema.prisma`)**  
   - Defined models (`User`, `ResidentRequest`, `Address`, `TimeSlot`) and enum (`RequestStatus`) to manage the application’s database schema.
   - Configured the PostgreSQL database as the datasource using environment variables.

5. **API Route for User Creation (`src/app/api/users/route.ts`)**  
   - Created a POST API route to add new users to the database.
   - Uses Prisma client to create a new user and return it in the response.
   - Includes error handling to log errors and return an error message if user creation fails.

6. **`layout.tsx`**  
   - Updated metadata for the app title and description.
   - Included Lato and Montserrat fonts with different weights, and customized `RootLayout` to apply them.

### How to Run the Application and Generate Prisma Migrations

To run the application and generate the Prisma migrations, follow these steps:

1. **Run Docker Compose**  
   - Ensure Docker is installed and running on your machine.
   - In the root of the project, execute:
     ```bash
     docker-compose up -d
     ```
   - This will start the PostgreSQL database defined in the `docker-compose.yml` file.

2. **Set Up Environment Variables**  
   - Create a `.env` file in the root directory with the following content:
     ```plaintext
     DATABASE_URL="postgresql://docker:docker@localhost:5432/apisippy"
     ```
   - Ensure that the `DATABASE_URL` matches the database credentials and settings in your `docker-compose.yml`.

3. **Generate Prisma Client**  
   - Run the following command to generate the Prisma client based on the schema:
     ```bash
     npm run prisma:generate
     ```

4. **Run Prisma Migrations**  
   - Apply the migrations to the PostgreSQL database:
     ```bash
     npm run prisma:migrate
     ```
   - This command will apply all pending migrations and set up the database schema as defined in `prisma/schema.prisma`.

5. **Start the Development Server**  
   - To start the Next.js development server, use:
     ```bash
     npm run dev
     ```
   - all set

### CODE REVIEW

File: src/app/api/users/route.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        firstName,
+        lastName,
+      },
+    });
+
+    return NextResponse.json(user, { status: 201 });
+  } catch (error) {
+    console.log(error)
+    return NextResponse.json({ error: "Erro ao criar usuário" }, { status: 500 });
+  }
```
Comment (n3ndor) [inline src/app/api/users/route.ts:23]: can we change to English error message ?

**Review by n3ndor — CHANGES_REQUESTED** [review]

File: src/app/page.tsx [hunk 2]
```diff
@@ -1,9 +1,44 @@
+'use client'
```
Comment (n3ndor) [inline src/app/page.tsx:1]: let's keep the main page file simple.
Please move this functions in the @/src/app/components/home/Home.tsx component

**Review by n3ndor — CHANGES_REQUESTED** [review]

File: src/app/api/users/route.ts (hunk 1 above)
Comment (Frankdias92) [reply src/app/api/users/route.ts:23]: For sure, sorry for that. I just created this function for test only. I will make the changes properly.

Thanks for observation Nandor.

**Comment by Frankdias92** [comment]
For the pull request, here’s a detailed description of each newly added and modified component:

1. **`Appointment` Component**: A user prompt that invites the user to set goals and create an appointment. Contains a button that routes to the `/calendar` page.

2. **`AppointmentHeaderStatus` Component**: Displays a progress header with steps, guiding the user through the form. Includes text and a progress bar, which dynamically updates based on the current stage.

3. **`AppointmentUserForm` Component**: User form with inputs for "Username" and "Full Name." It uses Zod for validation, displays error messages, and integrates `HeaderStatus` for a progress indicator.

4. **`ButtonText` Component**: Renders a stylized button with color options (`Green` or `Yellow`) and customizable text.

5. **`InputText` Component**: Customizable input field with an optional prefix label, error messages, and flexible text placeholder. Integrates `TextLabel` for consistency.

6. **`TextLabel` Component**: Provides text rendering with various font sizes and colors, maintaining a cohesive typography style.

7. **`WrapperContainer` Component**: A container to wrap form fields, aligning items and adding consistent spacing for better layout structure.

This PR adds a cohesive structure and styling consistency across the new user appointment flow, enhancing the UX/UI for a seamless onboarding experience.

**Review by davideastmond — APPROVED** [review]
This looks good 👍🏿

File: src/components/home/Home.tsx
Comment (Frankdias92) [inline src/components/home/Home.tsx:1]: I just see this import of now after the push, just remove please? <Appointment />

**Review by n3ndor — APPROVED** [review]

**Pull request merged**